### ISSUE TITLE
Unable to load app icon when just one app icon is present

### ISSUE THREAD

**Issue by foffux** [hons82/THMoreFromDeveloper#2]
Hello
Related to the title, I had that problem with your controller.
To solve that issue I went to Model class and I put following lines of code under "loadAppId" method:

``` objective-c
if ([json[@"results"] count] > 0) {
//tmp = [json[@"results"] subarrayWithRange:NSMakeRange(1, [json[@"results"] count]-1)];
tmp = json[@"results"];
}
```

I am wondering if there was a reason why you did not take app icon when it was the only one.

Other thing: think about a method like this in "Cell" class:

``` objective-c
- (void)setAppNameFont:(UIFont *)font withColor:(UIColor *)color
{
    [_appNameLabel setFont:font];
    [_appNameLabel setTextColor:color];
}
```

``` objective-c
- (void)setAppNameFont:(UIFont *)font withColor:(UIColor *)color;
```

to let user change font and color of app icon title. More in general, it would be better to give user more customization like the one I showed.

Do you need a pull request for this?

Please let me know what you think. Thank you.

**Comment by hons82** [comment]
Regarding the issue... I don't remember why I did that...
It might be that I call it with just AppIds in my [Apps](https://appsto.re/it/JoDs6.i), and that there is something in the jsonnthat I need to omit... 
However, I'll check that tonight.

For the customization... Why not. Actually in this control till now I put just what I needed. But I'm open to improvements. I'll check that too

**Comment by foffux** [comment]
Okay thank you

2015-05-26 16:51 GMT+02:00 Hannes notifications@github.com:

**Comment by hons82** [comment]
Now I finally found some time to check this things

1) Actually I think there is an understanding problem here. The model has these two methods

``` objective-c
- (void)loadDeveloperId:(NSString *)developerId {
    [self loadAppId:developerId iTunesRequestType:ITunesRequestTypeSoftware];
}

- (void)loadAppId:(NSString *)appId {
    [self loadAppId:appId iTunesRequestType:ITunesRequestTypeUnknown];
}
```

The first you need to feed with a [developerID (e.g. 833472034)](https://itunes.apple.com/lookup?country=IT&lang=en&id=833472034&entity=software), the second with an [appID (e.g. 833472031)](https://itunes.apple.com/lookup?country=IT&lang=en&id=833472031)

If you check the JSON behind the links you'll see that in the developer case the first result is an info about the developer which I currently just skip (this could be useful to show in some cases) and that's why I need to use the subarray in that case

2) your improvement... actually the variables image and label are accessible from outside, so basically you could just use them and apply font and color directly. But if you need such a method to access them you can just fork it, add the method, test it and send me a pull request to it.

Thanks for the help
~Hannes

**Comment by foffux** [comment]
Hello
Okay for point 2). As soon as I will have some spare time I will do it.
Cannot understand your answer at point 1). In fact, I am looking to show app image of my current apps. I looked into example but maybe I misunderstood it? I have to use my developer Id to load my app images instead directly my app ids? At the moment I tried to use app ids and in the case of just one app, I could not as I told you in my mail. 
Could you please clarify?
Thank you.

Inviato da iPad

**Comment by foffux** [comment]
Hi
I have just checked in my code:
I am using loadDeveloperId procedure. In fact I am filling in moreFromDeveloperIds = @[@"833472034"]; attribute. With this test developer id, it worked. But if I try to use mine, it fails.
What I noticed is that test developer id is 9 digits length instead my developer id is 8 digits length. Maybe something has changed in Apple side and you have to fix your controller? I found out my developer id looking into iTunesConnect web site, under Sales/Trends, clicking on Reports. Next to my own name, there is a number, starting with 8 but it is 8 digits length. It should be my developer id, since I read on stack overflow it should be there.
Thank you.
I will let you know if I will find out more.

**Comment by foffux** [comment]
Hi.
For the moment, I will use loadAppId procedure instead of loadDeveloperId procedure. I tried and it work also with only one app. Have you tested it for more than one app, right?
In my code I made the following changes:

``` javascript
//[self.model loadDeveloperId:moreFromDeveloperIds[0]];
[self.model loadAppIds:moreFromDeveloperIds];
```

Related to what I told you before, if you could investigate, it would be appreciated.
Thank you.

**Comment by hons82** [comment]
In short 
- `[self.model loadDeveloperId:@"123"]` has to be used with a developerID showing all APPs of a developer
- `[self.model loadAppIds:@["123",@"234"]]` can be used to show individual APPs and has to be used with appIDs

If you check the JSON of [this link](https://itunes.apple.com/lookup?country=IT&lang=en&id=833472034&entity=software) which is done using my developerID you'll c that there is an artistID(which is my developer ID) and every APP has its own trackID(which is the app ID)

Now it depends on you what you wanna show. All your apps or just a subset of APPs

**Comment by foffux** [comment]
Hello
your developer id is 833472034, so 9 digit legth. As I wrote you, mine is 8
and the it does not work. Where have you found yours? Maybe the one I
consider my developer id it is not.
Thank you.

2015-05-29 13:21 GMT+02:00 Hannes notifications@github.com:

**Comment by hons82** [comment]
That' s one of my Apps [MoneyBunny](https://itunes.apple.com/us/app/moneybunny/id833472031?mt=8)
On right top there is a link to ["more from this developer"](https://itunes.apple.com/us/artist/hannes-tribus/id833472034) if you click on that you'll c my apps and the link on top contains my id.

If you do that with one of your APPs you'll probably find your developerId

**Comment by foffux** [comment]
Hi
Yes, you are right
And I was able to let your controller work
But, now, another question arises: if I wanted all my app icons but not the
one related to the app where I am running your controller, how am I able to
do it? That is: an user is using myAppA and this is my first app. I f I
load all apps for a developer with your controller, I am able to see icon
for myAppA, but this is not important since I am using this app...Instead,
if I developed two apps, then will be interesting to show the app icon for
the app I am not actually using.
I do not know if I was able to explain you the actual problem.
Let me know.
Thanks.

2015-05-29 14:01 GMT+02:00 Hannes notifications@github.com:

**Comment by foffux** [comment]
Sorry for annoying you but I have another problem.
If I run
[self.model loadDeveloperId:moreFromDeveloperIds[0]];
in the app I have already released, JSON answer tells me "results count =
2" and I can see app Icon related to the app released.
But if I run same code in an app not yet released, JSON answer tells me
"results count = 1" and I cannot see app Icon related to that app I have
already released.
If I run, instead

[self.model loadAppIds:moreFromDeveloperIds];

with the app Id for the app already released, I can see its icon also in
the app I am testing and not released yet.

Why? I cannot understand...

2015-05-29 16:00 GMT+02:00 SuperFox foffux@gmail.com:

**Comment by foffux** [comment]
Hello
No comment about my two latest questions?
Thanks 

Inviato da iPhone

**Comment by hons82** [comment]
Currently we have a long weekend in Italy, so I'm not in the office for a couple of days :-)

I tested it last week with a couple of developerids and it worked for all of them
I tested it with a couple of appId arrays with different sizes and it worked too.

If you want to know what you'll get as result take the url that is constructed and check the results. That's what I get, and that's what I return in the controller. If there are icons not shown you'll probably find the answer there. If you really want me to check that then send me the ids that you use, but I'm not working on this before wednesday... Maybe even later... That's not my main work :-)

The only thing that I found in your posts that would be a real issue/improvement would be to have a flag or something to ignore the current App when loading all from developer, but I'd need to check if I can get the id from within the App... Maybe by comparing the bundle id... But I'm not sure about it

Have a nice (long) weekend
~hons
